.github/workflows/release.yml

```diff
@@ -16,33 +16,121 @@
 # specific language governing permissions and limitations
 # under the License.
 #
-name: Release Apache Spark
+
+# This workflow is intended for use in forked repositories
+# when manually dispatching this to create an RC.
+# To enable full release functionality, developers should manually configure
+# the following GitHub Secrets in their repository settings:
+#
+# - ASF_USERNAME:
+#     Your Apache Software Foundation (ASF) account ID.
+#
+# - ASF_PASSWORD:
+#     The password associated with your ASF account.
+#
+# - GPG_PRIVATE_KEY:
+#     Your GPG private key, exported using:
+#       gpg --armor --export-secret-keys ABCD1234 > private.key
+#     Ensure this key is registered with a public key server.
+#     For more details, refer to:
+#       https://spark.apache.org/release-process.html#preparing-gpg-key
+#
+# - GPG_PASSPHRASE:
+#     The passphrase for your GPG private key.
+#
+# This workflow supports dry runs by default. If the required GitHub Secrets are not provided,
+# only dry runs will be executed.
+#
+# In case something goes wrong during the process and a release candidate (RC) needs to be
+# cleaned up, follow these steps:
+#
+# 1. Revert the RC-related commits, such as:
+#    - "Preparing development version 3.5.7-SNAPSHOT"
+#    - "Preparing Spark release v3.5.6-rc1"
+#
+# 2. Delete the RC tag from the remote repository, for example:
+#    - git push --delete apache v3.5.6-rc1
+#
+# 3. Remove the RC artifacts from SVN:
+#    - RC=v3.5.6-rc1 && svn rm https://dist.apache.org/repos/dist/dev/spark/"${RC}"-bin/ -m "Removing RC artifacts."
+#    - RC=v3.5.6-rc1 && svn rm https://dist.apache.org/repos/dist/dev/spark/"${RC}"-docs/ -m "Removing RC artifacts."
+#
+# 4. Drop the staging repository if it exists (https://repository.apache.org/#stagingRepositories)
```

Comment:
it's OK to leave this one as well, as we have to clean up all the staging repositories during the RC stage when we finalize the release.